test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1859,9 +1860,20 @@ protected static <T> T copyInstance(
         try (BytesStreamOutput output = new BytesStreamOutput()) {
             output.setTransportVersion(version);
             writer.write(output, original);
-            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), namedWriteableRegistry)) {
-                in.setTransportVersion(version);
-                return reader.read(in);
+            if (randomBoolean()) {
```

Comment:
Nit - Is it worth adding some logging here so we can figure out what branch we're in if a test fails?

Reply:
I normally rely on the seed to understand why it has failed.